solutions/security/ai/connect-to-vLLM.md

```diff
@@ -0,0 +1,167 @@
+---
+applies_to:
+  stack: all
+  serverless:
+    security: all
+products:
+  - id: security
+  - id: cloud-serverless
+---
+
+# Connect to your own LLM using vLLM (air gapped environments)
+This page provides an example of how to connect to a self-hosted, open-source large language model (LLM) using the [vLLM inference engine](https://docs.vllm.ai/en/latest/) running in a Docker or Podman container. 
+
+Using this approach, you can power elastic's AI features with an LLM of your choice deployed and managed on infrastructure you control without granting external network access, which is particularly useful for air-gapped environments and organizations with strict network security policies. 
+
+## Requirements
+
+* Docker or Podman.
+* Necessary GPU drivers.
+
+## Server used in this example
+
+This example uses a GCP server configured as follows:
+
+* Operating system: Ubuntu 24.10
+* Machine type: a2-ultragpu-2g
+* vCPU: 24 (12 cores)
+* Architecture: x86/64
+* CPU Platform: Intel Cascade Lake
+* Memory: 340GB
+* Accelerator: 2 x NVIDIA A100 80GB GPUs
+* Reverse Proxy: Nginx
+
+## Outline
+The process involves four main steps:
+
+1. Configure your host server with the necessary GPU resources.
+2. Run the desired model in a vLLM container.
+3. Use a reverse proxy like Nginx to securely expose the endpoint to {{ecloud}}.
+4. Configure the OpenAI connector in your Elastic deployment.
+
+## Step 1: Configure your host server
+
+1. (Optional) If you plan to use a gated model (such as Llama 3.1) or a private model, create a [Hugging Face user access token](https://huggingface.co/docs/hub/en/security-tokens).
+  1. Log in to your Hugging Face account.
+  2. Navigate to **Settings > Access Tokens**.
+  3. Create a new token with at least `read` permissions. Save it in a secure location.
+2. Create an OpenAI-compatible secret token. Generate a strong, random string and save it in a secure location. You need the secret token to authenticate communication between {{ecloud}} and your Nginx reverse proxy.
+3. Install any necessary GPU drivers. 
+
+## Step 2: Run your vLLM container
+
+To pull and run your chosen vLLM image:
+
+1. Connect to your server using SSH.
+2. Run the following terminal command to start the vLLM server, download the model, and expose it on port 8000:
+
+```bash
+docker run --name Mistral-Small-3.2-24B --gpus all \
```

Comment:
is is something we will be able to update shortly? I mean we should avoid recommending `Mistral-Small-3.2-24B` as it has a lot of issues with Security Assistant tool calling

Reply:
I think it's going to be less confusing if we stay to the previous version and just update it with a new model, because the list of params depends on the model id